### PR TITLE
review: refactor: PrinterHelper to CommentHelper, LiteralHelper and OperatorHelper

### PR DESCRIPTION
**Pull request by pvojtechovsky**
First basic refactoring step needed to implement all the changes needed by #1494.

### CODE REVIEW

File: src/main/java/spoon/reflect/code/UnaryOperatorKind.java [hunk 1]
```diff
@@ -23,34 +23,47 @@
...
 	/**
 	 * Decrementation pre assignment.
 	 */
-	PREDEC, // -- _
+	PREDEC(true), // -- _
 	/**
 	 * Incrementation post assignment.
 	 */
-	POSTINC, // _ ++
+	POSTINC(false), // _ ++
```
Comment (monperrus) [inline src/main/java/spoon/reflect/code/UnaryOperatorKind.java]: I'd say we already have the postfix information in the naming convention. If it starts with POST it's postfixed, else if's prefixed. Correct?

**Comment by monperrus** [comment]
can the helper classes and methods be package visible? WDYT?

File: src/main/java/spoon/reflect/code/UnaryOperatorKind.java (hunk 1 above)
Comment (pvojtechovsky) [reply src/main/java/spoon/reflect/code/UnaryOperatorKind.java]: Will this be OK from your point of view?
```java
UnaryOperatorKind() {
  this.prefixOperator = name().startsWith("POST")==false;
}
```

**Comment by pvojtechovsky** [comment]
> can the helper classes and methods be package visible?

ok, no problem. Just when somebody will LATER write own printer, these helper classes will be very useful. But till then they can be package protected, so we can minimize API. I will do so.

**Comment by monperrus** [comment]
This code could even be put in one of the helper methods, in order not to touch and impact this core
model interface.

**Comment by monperrus** [comment]
I agree, it's important to be careful on the public API surface, so as to maximize our long-term API
stability.

If we create a new public class or method, then we have to document and test it extensively.

**Comment by pvojtechovsky** [comment]
> This code could even be put in one of the helper methods, in order not to touch and impact this core
model interface.

I guess you speak about `UnaryOperatorKind#isPrefixOperator()`, @monperrus why do you not like it? It is compatible is not?

**Comment by pvojtechovsky** [comment]
I made helpers package protected. It needed move of `CommentHelper` to `spoon.reflect.visitor`. 
Note: during later changes needed to reach #1494 there is probably necessary to make them public again. But we will see later.

I rolled back all the changes on operator enums. I created OperatorHelper methods
```java
public static boolean isPrefixOperator(UnaryOperatorKind o) {
	return isSufixOperator(o) == false;
}
public static boolean isSufixOperator(UnaryOperatorKind o) {
	return o.name().startsWith("POST");
}
```
It is ready for merge from my point of view

**Comment by pvojtechovsky** [comment]
> [WARNING] Failure to transfer commons-io:commons-io:2.5-RC4-SNAPSHOT/maven-metadata.xml from http://spoon.gforge.inria.fr/repositories/snapshots was cached in the local repository, resolution will not be reattempted until the update interval of gforge.inria.fr-snapshot has elapsed or updates are forced. Original error: Could not transfer metadata commons-io:commons-io:2.5-RC4-SNAPSHOT/maven-metadata.xml from/to gforge.inria.fr-snapshot 

CI has same problems...

**Comment by surli** [comment]
> CI has same problems...

Try to trigger it again: apparently there was some issue with Spoon website last days.

**Comment by pvojtechovsky** [comment]
CI finally built it. Please merge it or give me feedback, so I can continue with next steps ;-)

**Comment by monperrus** [comment]
Thanks Pavel!

**Comment by monperrus** [comment]
> I guess you speak about UnaryOperatorKind#isPrefixOperator(), @monperrus why do you not like it?

For the record: 1) because it impacts the core Spoon metamodel and this should be done super-carefully 2) it mixes the modeling responsibility and the pretty-printing responsibility

**Pull request merged**